crates/bifrost/src/loglets/local_loglet/log_store_writer.rs

```diff
@@ -161,27 +170,36 @@ impl LogStoreWriter {
         write_batch.put_cf(data_cf, &key.to_bytes(), data);
     }
 
-    fn commit(&mut self, opts: &LocalLogletOptions, write_batch: WriteBatch) {
-        let mut rocksdb_write_options = rocksdb::WriteOptions::new();
-        rocksdb_write_options.disable_wal(opts.rocksdb.rocksdb_disable_wal());
+    async fn commit(&mut self, opts: &LocalLogletOptions, write_batch: WriteBatch) {
+        let mut write_opts = rocksdb::WriteOptions::new();
+        write_opts.disable_wal(opts.rocksdb.rocksdb_disable_wal());
 
         trace!(
             "Committing local loglet current write batch: {} items",
             write_batch.len(),
         );
+        let result = self
+            .rocksdb
+            .write_batch(Priority::High, IoMode::Default, write_opts, write_batch)
+            .await;
 
-        if let Err(e) = self.db.write_opt(&write_batch, &rocksdb_write_options) {
+        if let Err(e) = result {
             error!("Failed to commit local loglet write batch: {}", e);
             self.send_acks(Err(Error::LogStoreError(e.into())));
             return;
         }
 
         if self.manual_wal_flush {
```

Comment:
Out of curiosity: How is the sync behavior if `manual_wal_flush` is set to false? After every write to RocksDB it will flush the WAL. Will it also be synced to disk or not?

Reply:
It seems that by default writes are not synced. For this to happen one needs to set `rocksdb::WriteOptions::set_sync(true)`. So we currently have a bit of mixed durability guarantees depending on whether `rocksdb_batch_wal_flushes` is enabled or not.

Reply:
Interesting observation is that RocksDB by default only flushes the WAL but does not sync to disc.

Reply:
@tillrohrmann and I agree with their default approach :)

Reply:
Maybe let's see what we put as the default once we have plugged things together and measured the impact of being conservative.